lib/galaxy/model/__init__.py

```diff
@@ -1157,7 +1177,7 @@ def is_hda(d):
     return isinstance( d, HistoryDatasetAssociation )
 
 
-class History( object, Dictifiable, UsesAnnotations, HasName ):
+class History( HasTags, Dictifiable, UsesAnnotations, HasName ):
```

Comment:
The reason it was done this was is because HasTags inherits from `object` and calls `super()` in order to let Dictifiable create the original dictionary (to which it adds the 'tags' key). I changed it in my latest commit to not inherit from `object` and use `super()` so will see if that breaks anything. Specifically I'm concerned about the 

This stuff certainly isn't straightforward to navigate in Python. If I understand it correctly, there *should* be a path from the subclass `to_dict()` up through the inheritance chain so that the *last* thing mentioned in the list is the one that actually creates the dictionary, everything else before it uses `super()` to get a dictionary and then adds to that dict.

My latest change has broken that principle so it might have broken the chain, since now HasTags and Dictifiable both have a `to_dict()` method that creates a dictionary. Not sure if the testing will catch this.